src/rust/engine/process_executor/src/lib.rs

```diff
@@ -0,0 +1,129 @@
+use std::io::Error;
+use std::process::Command;
+
+///
+/// A process to be executed.
+///
+/// # Example
+///
+/// ```
+/// use process_executor::ExecuteProcessRequest;
+/// let request = ExecuteProcessRequest {
+///     argv: vec!["/bin/bash".to_string(), "-c".to_string(), "/bin/echo Hello ${WHO}".to_string()],
+///     env: vec![("WHO".to_string(), "world".to_string())],
+/// };
+/// ```
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub struct ExecuteProcessRequest {
+    /// The arguments to execute. The first argument should be an absolute or relative path to the
+    /// binary to execute.
+    ///
+    /// No shell expansion or PATH lookup will be performed.
+    pub argv: Vec<String>,
+    /// The environment variables to set, in (key, value) pairs. Each key should appear at most one
+    /// time in the list.
+    ///
+    /// No other environment variables will be set for the execution (except possibly for an empty
+    /// PATH variable).
+    ///
+    /// Ideally this would be a HashMap, but that isn't hashable, and we need to implement Hash to
+    /// be in the node graph.
+    pub env: Vec<(String, String)>,
+}
+
+
+///
+/// The result of running a process.
+///
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct ExecuteProcessResult {
+    pub stdout: Vec<u8>,
+    pub stderr: Vec<u8>,
+    pub exit_code: i32,
+}
+
+///
+/// Runs a command.
+///
+pub fn run_command(req: ExecuteProcessRequest) -> Result<ExecuteProcessResult, Error> {
+    Command::new(&req.argv[0])
+        .args(&req.argv[1..])
+        .env_clear()
+        // It would be really nice not to have to manually set PATH but this is sadly the only way
+        // to stop automatic PATH searching.
+        .env("PATH", "")
```

Comment:
It seems the order should be `.envs(req.env).env("PATH", "")` to ensure any PATH in req.env doesn't sneak in.

Reply:
If people want to set PATH, they can. There are many third-party scripts (like the cloc we use :)) which depend on PATH for things it execs.

Updated docs to state that "No PATH lookup will be performed unless a PATH environment variable is specified."

I'm kind of unhappy that Rust doesn't have an option to turn PATH resolution off.